### PR TITLE
Implement AWS IAM authentication

### PR DESCRIPTION
**Pull request by karenc-bq**
### Summary

Implement AWS IAM authentication

#### Added
- AWS IAM authentication
- Integration tests

### Additional Reviewers

@sergiyvamz 
@hsuamz 
@ColinKYuen

### CODE REVIEW

File: pgjdbc/src/main/java/org/postgresql/core/v3/plugins/AuthenticationPlugin.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
+ * AWS JDBC Driver for PostgreSQL
+ * Copyright Amazon.com Inc. or affiliates.
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.core.v3.plugins;
+
+public interface AuthenticationPlugin {
```
Comment (sergiyvamz) [inline pgjdbc/src/main/java/org/postgresql/core/v3/plugins/AuthenticationPlugin.java:9]: Let's add "I" to interface name: IAuthenticationPlugin.

File: pgjdbc/src/test/java/software/aws/rds/jdbc/postgresql/ca/AwsIamAuthenticationIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.postgresql.util.PSQLException;
+
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+@Disabled
+public class AwsIamAuthenticationIntegrationTests {
+
+  private static final String CONNECTION_STRING_PREFIX = "jdbc:postgresql:aws://";
```
Comment (sergiyvamz) [inline pgjdbc/src/test/java/software/aws/rds/jdbc/postgresql/ca/AwsIamAuthenticationIntegrationTests.java:16]: "jdbc:postgresql:aws://" -> "jdbc:postgresql://" ?
Let's make test more generic, not depend on AWS driver

**Review by sergiyvamz — APPROVED** [review]

File: pgjdbc/src/main/java/org/postgresql/core/v3/plugins/AuthenticationPlugin.java (hunk 1 above)
Comment (karenc-bq) [reply pgjdbc/src/main/java/org/postgresql/core/v3/plugins/AuthenticationPlugin.java:9]: Added.

File: pgjdbc/src/test/java/software/aws/rds/jdbc/postgresql/ca/AwsIamAuthenticationIntegrationTests.java (hunk 2 above)
Comment (karenc-bq) [reply pgjdbc/src/test/java/software/aws/rds/jdbc/postgresql/ca/AwsIamAuthenticationIntegrationTests.java]: Makes sense. Updated.

**Pull request merged**